File: BibliotecaPY/Sociclass.py
from Llibreclass import Llibre

class Soci:
    def __init__(self, nom, cognom, id):
        self.nom = nom
        self.cognom = cognom
        self.id = id

    def demanar_prestec(self, data_prestec):
        print(f"El soci {self.nom} {self.cognom} (id: {self.id}) ha demanat un préstec.")

    def retornar_prestec(self):
        print(f"El soci {self.nom} {self.cognom} (id: {self.id}) ha retornat un préstec.")

    def mostrar_info(self):
        print(f"Nom: {self.nom}, Cognom: {self.cognom}, ID de soci: {self.id}")
    

File: Kotlin/BibliotecaKT/Sociclass.kt
package Biblioteca

class Soci(val nom: String, val cognom: String, val numeroSoci: Int) {

    fun sol·licitarPrestec(llibre: Llibre, dataPrestec: String) {
        println("Sol·licitud de préstec feta per $nom $cognom, Número de Soci: $numeroSoci")
        val prestec = Prestec(llibre, this)
        prestec.registrarPrestec(dataPrestec)
    }

    fun retornarPrestec(llibre: Llibre) {
        println("Devolució de préstec feta per $nom $cognom, Número de Soci: $numeroSoci")
        val prestec = Prestec(llibre, this)
        prestec.retornarPrestec()
    }

    fun informacio() {
        println("Nom: $nom, Cognom: $cognom, Número de Soci: $numeroSoci")
    }
}
